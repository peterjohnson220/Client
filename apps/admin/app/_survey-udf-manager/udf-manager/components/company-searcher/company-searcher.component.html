<div class="form-container container d-flex">
  <h6 class="select-company mt-2" *ngIf="!selectedCompany">Please select a company</h6>
  <kendo-combobox
    class="form-control"
    [attr.data-qa-id]="'txt-company'"
    [data]="companiesList"
    [placeholder]="'Select Company'"
    [textField]="'Name'"
    [valueField]="'CompanyId'"
    [filterable]="true"
    [valuePrimitive]="false"
    [loading]="loadingCompaniesList"
    (filterChange)="handleCompanyListFilter($event)"
    (valueChange)="setSelectedCompany($event)">
    <ng-template kendoComboBoxItemTemplate let-dataItem>
      <span>{{ dataItem.Name }} ({{ dataItem.CompanyId }})</span>
    </ng-template>
  </kendo-combobox>
</div>
