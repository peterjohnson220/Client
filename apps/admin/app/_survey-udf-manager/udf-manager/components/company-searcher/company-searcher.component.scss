h6.select-company {
  min-width: 155px;
}