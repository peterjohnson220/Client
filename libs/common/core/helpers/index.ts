export * from './kendo-grid-filter.helper';
