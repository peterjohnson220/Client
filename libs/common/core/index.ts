export * from './pipes';
export * from './helpers';
