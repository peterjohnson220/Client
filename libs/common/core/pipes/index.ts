export * from './humanize-number.pipe';
export * from './truncate-after.pipe';
