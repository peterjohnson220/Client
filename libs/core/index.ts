export * from './helpers';
export * from './pipes';
export * from './common.module';
