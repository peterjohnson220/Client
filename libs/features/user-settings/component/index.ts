export * from './auto-share-modal';
export * from './auto-shared-users';
export * from './shareable-user';
