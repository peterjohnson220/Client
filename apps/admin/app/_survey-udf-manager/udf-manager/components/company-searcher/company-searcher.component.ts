import { Component, EventEmitter, Input, Output } from '@angular/core';

import { BehaviorSubject } from 'rxjs';
import { debounceTime, distinctUntilChanged } from 'rxjs/operators';

import { CompanyBaseInformation } from 'libs/models/company';

const take = 100;

@Component({
  selector: 'pf-company-searcher',
  templateUrl: './company-searcher.component.html',
  styleUrls: ['./company-searcher.component.scss']
})
export class CompanySearcherComponent {

  @Input() loadingCompaniesList: boolean;
  @Input() companiesList: CompanyBaseInformation[];
  @Input() selectedCompany: CompanyBaseInformation;

  @Output() selectCompany = new EventEmitter<CompanyBaseInformation>();
  @Output() unselectCompany = new EventEmitter();
  @Output() filterCompanies = new EventEmitter<{ searchTerm: string, take: number }>();

  searchTermChanged$ = new BehaviorSubject<string>('');

  constructor() {
    this.searchTermChanged$.pipe(
      debounceTime(400),
      distinctUntilChanged()
    )
    .subscribe(searchTerm => this.filterCompanies.emit({ searchTerm, take }));
  }

  handleCompanyListFilter(searchTerm: string) {
    this.searchTermChanged$.next(!searchTerm ? '' : searchTerm);
  }

  setSelectedCompany(company: CompanyBaseInformation) {
    if (company) {
      this.selectCompany.emit(company);
    } else {
      this.unselectCompany.emit();
    }
  }

}
