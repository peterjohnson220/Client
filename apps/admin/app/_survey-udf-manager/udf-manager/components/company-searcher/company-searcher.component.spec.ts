import { NO_ERRORS_SCHEMA } from '@angular/core';
import { ComponentFixture, TestBed, waitForAsync } from '@angular/core/testing';

import { CompanySearcherComponent } from './company-searcher.component';

describe('CompanySearcherComponent', () => {
  let component: CompanySearcherComponent;
  let fixture: ComponentFixture<CompanySearcherComponent>;

  beforeEach(waitForAsync(() => {
    TestBed.configureTestingModule({
      declarations: [CompanySearcherComponent],
      schemas: [NO_ERRORS_SCHEMA]
    });
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(CompanySearcherComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should show an instructional message when no company is selected', () => {
    component.selectedCompany = null;
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

  it('should hide the instructional message when a company is selected', () => {
    component.selectedCompany = { CompanyId: 567 } as any;
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });
});
