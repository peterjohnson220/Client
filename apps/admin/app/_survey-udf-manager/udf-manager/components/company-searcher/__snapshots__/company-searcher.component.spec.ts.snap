// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompanySearcherComponent should hide the instructional message when a company is selected 1`] = `
<pf-company-searcher
  filteredCompaniesList={[Function Array]}
  selectCompany={[Function EventEmitter]}
  selectedCompany={[Function Object]}
  unselectCompany={[Function EventEmitter]}
>
  <div
    class="form-container container d-flex"
  >
    
    <kendo-combobox
      class="form-control"
      data-qa-id="txt-company"
    >
      
    </kendo-combobox>
  </div>
</pf-company-searcher>
`;

exports[`CompanySearcherComponent should show an instructional message when no company is selected 1`] = `
<pf-company-searcher
  filteredCompaniesList={[Function Array]}
  selectCompany={[Function EventEmitter]}
  selectedCompany="null"
  unselectCompany={[Function EventEmitter]}
>
  <div
    class="form-container container d-flex"
  >
    
    <h6
      class="select-company mt-2"
    >
      Please select a company
    </h6>
    <kendo-combobox
      class="form-control"
      data-qa-id="txt-company"
    >
      
    </kendo-combobox>
  </div>
</pf-company-searcher>
`;
